Java/math/polygonal_number.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * A polygonal number is the number of equally spaced dots needed to draw a polygon.
+ * It is given by the formula:
+ *                                  Z = k/2 [2 + (k - 1)d]
+ *        where k is the kth term and d is (number of vertices in the polygon - 2)
+ *        
+ * This program generates the first n polygonal numbers with input number of vertices.
+ */
+import java.util.*;
+
+public class polygonal_number
+{
+    void main()
```

Comment:
void main() in java?

Reply:
I have formatted the code as per your style guidelines. It says to put the driver code in main function and the main logic in another function.